### PR TITLE
Declare an install target for a mini library for unit test support, so roottest can be built standalone

### LINKED ISSUE #9507: Not possible anymore to build `roottest` standalone
In the past, it was possible to build the [roottest suite](https://github.com/root-project/roottest) standalone, just having ROOT sourced in your environment. This possibility is also advertised in the README.md of roottest.

After https://github.com/root-project/root/commit/56c6ace5340003a9135d565f4d5fa68be67aa0d5, it is not possible anymore to do that, because the unit tests depend on the `ROOTUnitTestSupport` library which is only available at ROOT compile time.

Maybe a possible solution is to install the `ROOTUnitTestSupport` library as part of ROOT? I don't know what's the best solution here, so I'm only opening an issue for now. But I think this need to be fixed before the next ROOT release.

### PR DESCRIPTION
**Pull request by hageboeck**
In order to be able to build unit test support also outside of a ROOT
build (e.g. for standalone roottest or for user tests), the unit test
support library needs to be visible outside of ROOT. Therefore, it is
moved to `core/testsupport`, and now installed and exported under the
name `ROOT::TestSupport` when testing is enabled.

A warning message was added to `ROOT_ADD_GTEST` in case this library isn't
found when a new gtest is defined.

Fix #9507.

## For reviewers
- [x] I'm not happy with the naming. The folder is `core/test_support`, the library `ROOT::ROOTTestSupport`, but the namespace and the header are historically named `ROOTUnitTestSupport`. Which one?

### CODE REVIEW

**Review by Axel-Naumann — COMMENTED** [review]
From a non-expert (CMake or this testing code) point of view this looks great, thanks @hageboeck !

File: core/test_support/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# ROOTTestSupport: the unit-test support library
+
+This library supports ROOT's unit tests. It implements two main functions
+1. It provides a static library target `ROOTTestSupport`. All google tests that are defined using `ROOT_ADD_GTEST` will be linked against this target. When a tests executable starts up, this will install a ROOT message handler that intercepts all messages / diagnostics. If a message with severity > kInfo is issued, this message handler will generate a test failure.
+2. It provides tools to declare in code when messages are expected during a test. For this
+   1. Include the header ROOTUnitTestSupport.h.
+   2. Declare a RAII object that temporarily replaces the message handler from 1. Register the expected messages to this object, so it can check that they are indeed sent.
+      This could look as follows:
+      ```c++
```
[the remark this answers is not in the document]
Comment (hageboeck) [reply core/test_support/README.md:9]: I cannot? Check this:
https://github.com/hageboeck/root/tree/unit_test_support/core/test_support

[the remark this answers is not in the document]
Comment (Axel-Naumann) [reply core/test_support/README.md:9]: OK let's see :-)

**Comment by vgvassilev** [comment]
I think we want to test code in core/foundation which is the most restrictive place in ROOT - eg we can run with rootcling_stage1, etc. I think we can move the library there to avoid introducing incompatible changes.

Maybe we can avoid the new library situation and conditionally add the testing header/source files to an existing library, eg libFoundation?

**Comment by hageboeck** [comment]
> I think we want to test code in core/foundation which is the most restrictive place in ROOT - eg we can run with rootcling_stage1, etc. I think we can move the library there to avoid introducing incompatible changes.
> 
> Maybe we can avoid the new library situation and conditionally add the testing header/source files to an existing library, eg libFoundation?

That should be possible. What do you think @Axel-Naumann ?

**Comment by hageboeck** [comment]
> That should be possible. What do you think @Axel-Naumann ?

2nd thought:
It depends on `gtest`, so better not to put it in any existing lib.

**Comment by guitargeek** [comment]
Hi @hageboeck, thanks a lot for the quick fix! About your naming problem:

> I'm not happy with the naming. The folder is `core/test_support`, the library `ROOT::ROOTTestSupport`, but the namespace and the header are historically named `ROOTUnitTestSupport`. Which one?

This would be my reasoning:
0) Assuming that no user uses the test support library, only us in `root-project` repos so we can do namespace changes 
1) Looking at the the other libraries in `core`, the folder names are all concatenated without any delimiter (`rootcling_stage` is the only exception).
2) Historical names are not important anymore (at least for the git history) because the files are move anyway.
3) If you propose to name the library `ROOT::ROOTTestSupport`, is `ROOT` not redundant?

So I would go for folder `core/testsupport`, both library and namespace `ROOT::TestSupport`, and the header file moves from `ROOTUnitTestSupport.h` to `inc/ROOT/TestSupport.hxx`. This would follow the example set by [`core/foundation`](https://github.com/root-project/root/tree/master/core/foundation), which I consider as a modern ROOT library where everything is done correctly.

**Comment by hageboeck** [comment]
> So I would go for folder `core/testsupport`, both library and namespace `ROOT::TestSupport`, and the header file moves from `ROOTUnitTestSupport.h` to `inc/ROOT/TestSupport.hxx`.

This is *exactly* what I discussed with Axel in the mean time. Excellent that you think along the same lines. Working on it ...

**Comment by hageboeck** [comment]
@vgvassilev @Axel-Naumann 
I think that's it. I will squash commits 1 and 2, since the renaming creates a lot of noise that you don't need to review. Check the first commit, say if you are happy, and then I'll squash the renaming of the namespace into that first commit, so the master history is always in a state that will build.

**Review by Axel-Naumann — APPROVED** [review]
Thanks for doing that!

**Pull request merged**